code/datums/components/hallucinations.dm

```diff
@@ -549,9 +600,138 @@ ABSTRACT_TYPE(/datum/component/hallucination)
 			for(var/mob/O in oviewers(world.view , my_target))
 				boutput(O, SPAN_ALERT("<B>[my_target] stumbles around.</B>"))
 
+/obj/fake_singulo
+	name = "gravitational singularity"
+	desc = "Perhaps the densest thing in existence, except for you."
+	icon = null
+	icon_state = null
+	density = 0
+	var/list/my_image_overrides = list()
+	var/list/eaten_atoms = list()
+	var/mob/my_target = null
+	var/spaget_count = 0
+	var/right_spinning = 0
+
+/obj/fake_singulo/New(var/loc, var/mob/target, var/ttl = 60 SECONDS)
+	..()
+	SPAWN(ttl)
+		qdel(src)
+	src.right_spinning = prob(50)
+	src.my_target = target
+	var/image/client_image = image(icon = 'icons/effects/64x64.dmi', icon_state = "whole", loc = src)
+	client_image.override = TRUE
+	client_image.plane = PLANE_DEFAULT_NOWARP
+	var/image/lense = image(icon='icons/effects/overlays/lensing.dmi', icon_state="lensing_med_hole", pixel_x = -208, pixel_y = -208, loc = src)
+	lense.plane = PLANE_DISTORTION
+	lense.blend_mode = BLEND_OVERLAY
+	lense.appearance_flags = RESET_ALPHA | RESET_COLOR
+	src.my_image_overrides += client_image
+	src.my_image_overrides += lense
+	target << client_image
+	target << lense
+	process()
+
+/obj/fake_singulo/disposing()
+	if(src.my_image_overrides && my_target?.client)
+		for(var/client_image in src.my_image_overrides)
+			my_target.client?.images -= client_image
+			qdel(client_image)
+	qdel(src.my_image_overrides) //not really necessary, but why not
+	qdel(src.eaten_atoms)
+	. = ..()
 
 
+/obj/fake_singulo/proc/process()
+	var/target_dist = get_dist(src, src.my_target)
+	if(target_dist > 20) //if offscreen by a good amount
+		qdel(src)
+		return
+	else if(target_dist < 6)
+		//oh no you're being pulled into the singularity!
+		if(prob(50))
+			src.my_target.step_towards_movedelay(src) //respect standard movements, this isn't actual gravity, you're just stepping cos you're hallucinating
+
+	//otherwise, we're pretending to be a singularity
+	if(prob(30))
+		step(src, pick(cardinal))
+	else
+		step_towards(src, src.my_target) //oh god it's chasing me!
+
+	//"eat" stuff by overriding its icon with a blank icon state
+	for (var/X in range(3, src))
+		if (!X)
+			continue
+		if (X == src)
+			continue
+		var/atom/A = X
```

Comment:
Why not just do `for (var/atom/A` then?

Reply:
copying from old code without giving it much thought :D I've made the same change to the singularity code also